src/estimagic/optimization/tranquilo/get_component.py

```diff
@@ -143,6 +145,12 @@ def _get_valid_options(
     _default_options = {} if default_options is None else default_options
     _user_options = {} if user_options is None else user_options
 
+    if not isinstance(_user_options, dict):
+        _user_options = user_options._asdict()
+
+    if not isinstance(_default_options, dict):
+        _default_options = default_options._asdict()
+
```

Comment:
I think we should use `update_option_bundle` function here to get the automatic type checks and conversions for free. This might require moving it to `options.py` to avoid circular imports.